src/app/component/SearchOverlay/SearchOverlay.component.js

```diff
@@ -139,7 +139,7 @@ export default class SearchOverlay extends PureComponent {
             <p
               block="SearchOverlay"
               elem="Criteria"
-              mods={ { isVisible: !!searchCriteria } }
+              mods={ { isVisible: !!searchCriteria.trim() } }
```

Comment:
Theoretically, if there is `SearchOverlay.container.js` you can process `searchCriteria` trimming there.

Reply:
There's shouldn't be any problem because we don't modify user input, we just not making any actions on empty input.